### PR TITLE
implement product service

### PR DESCRIPTION
**Pull request by MishaHnatyshyn**
Service is done, FE is done

Additional scope - async/await, es6 modules, webpack, unit tests, modularity, error scenarios handling.

Link to product-service API: https://e0y68fypfe.execute-api.us-east-1.amazonaws.com/dev/products

Link to FE App MR: https://github.com/MishaHnatyshyn/nodejs-aws-fe/pull/2

`
200 OK - GET /products
{
  status: "SUCCESS",
  data: [{
    count: number,
    description: string,
    id: string,
    price: number,
    title: string,
    author: string,
    pagesCount: number,
    imageUrl: string,
  }, ....]
}
`

`
200 OK - GET /products/{productId}
{
  status: "SUCCESS",
  data: {
    count: number,
    description: string,
    id: string,
    price: number,
    title: string,
    author: string,
    pagesCount: number,
    imageUrl: string,
  }
}
`

`
404 Not Found - GET /products/{productId}
{
  status: "FAILURE",
  message: "Sorry, can't find a book with id {productId}. Try using another id!"
}
`

`
500 Internal Server Error - GET both /products/{productId} and /products
{
  status: "FAILURE",
  message: "Something went wrong :("
}
`

### CODE REVIEW

**Comment by aliakseikrauchanka** [comment]
Sorry what is the goal of adding `status` field in the response. Is it not enough to use HTTP status codes instead?

File: product-service/handlers/getProductsById.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  formDefaultServerErrorResponse,
+  formFailureResponseBody,
+  formResponse,
+  formSuccessResponseBody
+} from '../utils/response';
+import { Book } from '../types/book.interface';
+import { HttpResponseStatus } from '../types/HttpResponseStatus.enum';
+import BookNotFoundError from '../services/book/bookNotFound.error';
+
+const getProductsById: APIGatewayProxyHandler = async (event, _context) => {
```
Comment (aliakseikrauchanka) [inline product-service/handlers/getProductsById.ts:13]: I believe `async` keyword is not needed here

Comment (aliakseikrauchanka) [reply product-service/handlers/getProductsById.ts:13]: Ok I believe you'we added `async` and `await` with intention to manage some async later and not just to complete task. Let's go with this

File: product-service/package.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "product-service",
+  "version": "1.0.0",
+  "description": "Serverless webpack example using Typescript",
+  "main": "handler.js",
+  "scripts": {
+    "deploy": "sls depoloy",
```
Comment (aliakseikrauchanka) [inline product-service/package.json]: Small typo here: `deploy`

**Comment by aliakseikrauchanka** [comment]
Nice!

**Comment by MishaHnatyshyn** [comment]
> Sorry what is the goal of adding `status` field in the response. Is it not enough to use HTTP status codes instead?

It's just some additional information about a response.

**Pull request merged**